azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1900,6 +1955,12 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
     if (project == null) {
       return;
     }
+    // fail the upload if the project is CRT locked and the user is not the CRT user
+    if (project.isCRTLocked() && !user.getUserId().equals(CRT_USER)) {
+      registerError(ret, "Installation Failed. Project '" + projectName + " has CRT LOCK on. "
```

Comment:
Authorization to upload operation is handled in `validateUploadAndGetProject()`. This seems to conflict with the results of that method as it's removing permission to upload to Azkaban admins. Is that the intention? If yes, why?

Reply:
This intention is to reduce human effect on CICD enabled projects. Azkaban admin would have the adhoc feature flag enabled. Once it is enabled, normal validation would apply on that upload.